lib/fog/openstack/models/compute/server.rb

```diff
@@ -135,16 +135,24 @@ def floating_ip_addresses
           all_floating.empty? ? manual : all_floating
         end
 
-        alias_method :public_ip_addresses, :floating_ip_addresses
+        def public_ip_addresses
+          if floating_ip_addresses.empty?
+            addresses.find_all{|s| s[0] =~ /public/i}.map{|a| a[1][0]['addr']}
```

Comment:
```public_ip_addresses``` are not the same as ```floating_ip_addresses```

If you have attached floating ip addresses to the vm, you can assume that those are public.

If you don't use floating_ip_addresses and instead have networks with /public/i in the name, use those instead.

Otherwise the logic is the same.

I'm not yet familiar with the testing framework, but with some quick direction / pointers I could try to code something up.